crates/sui-indexer/migrations/2022-12-01-034426_objects/up.sql

```diff
@@ -1,32 +1,99 @@
 CREATE TYPE owner_type AS ENUM ('address_owner', 'object_owner', 'shared', 'immutable');
+CREATE TYPE change_type AS ENUM ('new', 'modify', 'delete');
 
-CREATE TABLE objects (
-    id BIGSERIAL PRIMARY KEY,
-    object_id VARCHAR(255) NOT NULL,
-    version BIGINT NOT NULL,
-
+CREATE TABLE objects
+(
+    epoch                  BIGINT       NOT NULL,
+    checkpoint             BIGINT       NOT NULL,
+    object_id              address PRIMARY KEY,
+    version                BIGINT       NOT NULL,
+    object_digest          base58digest NOT NULL,
     -- owner related
-    owner_type owner_type NOT NULL,
+    owner_type             owner_type   NOT NULL,
     -- only non-null for objects with an owner,
     -- the owner can be an account or an object. 
-    owner_address VARCHAR(255),
+    owner_address          address,
     -- only non-null for shared objects
     initial_shared_version BIGINT,
-
-    package_id TEXT NOT NULL,
-    transaction_module TEXT NOT NULL,
-    object_type TEXT,
-
-    -- status can be CREATED, MUTATED or DELETED.
-    object_status VARCHAR(255) NOT NULL
+    previous_transaction   base58digest NOT NULL,
+    package_id             address      NOT NULL,
+    transaction_module     VARCHAR      NOT NULL,
+    object_type            VARCHAR      NOT NULL
 );
-
 CREATE INDEX objects_owner_address ON objects (owner_address);
 CREATE INDEX objects_package_id ON objects (package_id);
-CREATE INDEX objects_object_id ON objects (object_id);
 
-CREATE TABLE object_logs (
-    last_processed_id BIGINT PRIMARY KEY
-);
+CREATE TABLE objects_history
+(
+    epoch                  BIGINT       NOT NULL,
+    checkpoint             BIGINT       NOT NULL,
+    object_id              address      NOT NULL,
+    version                BIGINT       NOT NULL,
+    object_digest          base58digest NOT NULL,
+    owner_type             owner_type   NOT NULL,
+    owner_address          address,
+    initial_shared_version BIGINT,
+    previous_transaction   base58digest NOT NULL,
+    package_id             address      NOT NULL,
+    transaction_module     VARCHAR      NOT NULL,
+    object_type            VARCHAR      NOT NULL,
+    object_status          change_type  NOT NULL,
+    CONSTRAINT objects_history_pk PRIMARY KEY (epoch, object_id, version)
+) PARTITION BY RANGE (epoch);
+-- TODO: Add trigger to automatically create partitions at new epoch when we have Epoch Table
```

Comment:
Is my understanding correct that this needs to be done before the next release?

Reply:
to double check, this table can be used for txn query via mutated objects, right?

Reply:
ya this table can be use like ordinary table, e.g. you can query the change history of object using 
```
SELECT checkpoint, version, object_digest, previous_transaction, object_status
FROM objects_history
WHERE object_id = '0x32f7eb342bd16ab03270544bd82dfae52699955a6c01c0560a5a38608a34de1c'
```
result:
<img width="1388" alt="image" src="https://user-images.githubusercontent.com/1888654/223291993-9df33583-5ea4-42f9-80e5-7e3f470e9756.png">

Reply:
yes, this is taken from `sui_types::Object::pervious_transaction`